examples/js/loaders/STLLoader.js

```diff
@@ -200,37 +200,63 @@ THREE.STLLoader.prototype = {
 
 	parseASCII: function ( data ) {
 
-		var geometry, length, patternFace, patternNormal, patternVertex, result, text;
+		var geometry, length, patternFace, patternNormal, patternVertex, result, text, vertexCountPerFace, normalCountPerFace, faceCounter;
+		
+		faceCounter = vertexCountPerFace = normalCountPerFace = 0;
 		geometry = new THREE.BufferGeometry();
 		patternFace = /facet([\s\S]*?)endfacet/g;
+		
+		var patternFloat = /[\s]+([+-]?(?:\d+.\d+|\d+.|\d+|.\d+)(?:[eE][+-]?\d+)?)/.source;
+		patternNormal = new RegExp(''
+			+ 'normal'
+			+ patternFloat
+			+ patternFloat
+			+ patternFloat
+			, 'g'
+		);
+		
+		patternVertex = new RegExp(''
+			+ 'vertex'
+			+ patternFloat
+			+ patternFloat
+			+ patternFloat
+			, 'g'
+		);
 
 		var vertices = [];
 		var normals = [];
 
 		var normal = new THREE.Vector3();
 
 		while ( ( result = patternFace.exec( data ) ) !== null ) {
+			vertexCountPerFace = normalCountPerFace = 0;
 
 			text = result[ 0 ];
-			patternNormal = /normal[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+/g;
 
 			while ( ( result = patternNormal.exec( text ) ) !== null ) {
 
 				normal.x = parseFloat( result[ 1 ] );
-				normal.y = parseFloat( result[ 3 ] );
-				normal.z = parseFloat( result[ 5 ] );
-
+				normal.y = parseFloat( result[ 2 ] );
+				normal.z = parseFloat( result[ 3 ] );
+				normalCountPerFace++;
 			}
 
-			patternVertex = /vertex[\s]+([\-+]?[0-9]+\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+[\s]+([\-+]?[0-9]*\.?[0-9]*([eE][\-+]?[0-9]+)?)+/g;
-
 			while ( ( result = patternVertex.exec( text ) ) !== null ) {
 
-				vertices.push( parseFloat( result[ 1 ] ), parseFloat( result[ 3 ] ), parseFloat( result[ 5 ] ) );
+				vertices.push( parseFloat( result[ 1 ] ), parseFloat( result[ 2 ] ), parseFloat( result[ 3 ] ) );
 				normals.push( normal.x, normal.y, normal.z );
-
+				vertexCountPerFace++;
 			}
-
+					
+			// Every face have to own ONE valid normal
+			if (normalCountPerFace !== 1) {
+				throw 'Something isn\' right with the normal of face number ' + faceCounter;
+			}
+			// Every face have to own THREE valid vertices
+			if (vertexCountPerFace !== 3) {
+				throw 'Something isn\' right with the vertices of face number ' + faceCounter;
+			}
+			faceCounter++;
 		}
 
```

Comment:
Only some little notes on this file:

1.	`length` (in the past defined in line 203) could be removed since it is never used 
2.	You are throwing a string (which is possible in JavaScript), but common practice in three.js seems to be to throw a `new Error`
3.	Small typo (missing “t”) in isn\’t (2x)
4.	Shouldn’t it be better “Each face has” instead of “Every face have” (2x)?
5.	Perhaps you should check the code with http://zz85.github.io/mrdoobapproves/, so that it matches @mrdoob’s code style guide (there are some missing spaces and some empty lines with whitespace)

And one question: two other files that have no relation to this issue are indicated as changed, too. I don’t know what this means and if it could be a problem.